### PR TITLE
MRG, MAINT: Add a roadmap

### PR DESCRIPTION
**Pull request by larsoner**
This does not have to be an exhaustive list for now, but it would be nice to have. It will give us some guidelines, and for things like GSoC we might be able to say "look at the roadmap list".

I plan to add more tomorrow (and link to issues, etc.), but if people have immediate ideas feel free to comment.

### CODE REVIEW

**Comment by larsoner** [comment]
@drammock can you look at the rendered output and see if there are things you want to add/reorder/whatever?

https://17678-1301584-gh.circle-artifacts.com/0/dev/overview/roadmap.html

Feel free to push directly to my PR. Once you're happy we can have @agramfort take a look and merge quickly. This only needs to be a first draft, not a complete one. We can add more later, just want to make sure that these are all at least reasonable goals that are sufficiently interesting to enough people to be included in our roadmap.

After this is in, I'll probably clean up the Wiki to tell GSoC people to look at this list, etc. Probably once GSoC ramps up again I'll open another PR to talk about expected difficulty, appropriateness for GSoC, etc.

**Comment by drammock** [comment]
I pushed a few wording / spelling tweaks to the roadmap, otherwise LGTM, ready for @agramfort review/merge

File: doc/overview/roadmap.rst [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+
+3D visualization
+^^^^^^^^^^^^^^^^
+
+Historically we have used Mayavi for 3D visualization, but have faced
+limitations and challenges with it. We should work to use some other backend
+(e.g., PyVista) to get major improvements, such as:
+
+1. Proper notebook support (XTK through pyvtk?)
```
Comment (agramfort) [inline doc/overview/roadmap.rst]:
```suggestion
1. Proper notebook support (through vtkjs)
```

I don't have a clear idea what vtkjs option will win the war of standard in the community

File: doc/overview/roadmap.rst [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+conjunction with FieldTrip. Example of NutmegTrip's source-time-frequency mode
+in action (click for link to YouTube):
+
+.. image:: https://i.ytimg.com/vi/xKdjZZphdNc/maxresdefault.jpg
+   :target: https://www.youtube.com/watch?v=xKdjZZphdNc
+   :width: 50%
+
+
+Cluster computing
+^^^^^^^^^^^^^^^^^
```
Comment (agramfort) [inline doc/overview/roadmap.rst]:
```suggestion
^^^^^^^^^^^^^^^^^

```

File: doc/overview/roadmap.rst [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+  - In code, switch to cloud computing rather than local computing via a simple
+    change of n_jobs parameter, and/or context manager like with::
+
+        with use_dask(...):
+           ...
+
+- Develop a (short as possible) example that shows people how to run a minimal
+  task remotely, including setting up access, cluster, nodes, etc.
+- Adapt
+  MNE-biomag-group-demo code to use cloud computing (optionally, based on
```
Comment (agramfort) [inline doc/overview/roadmap.rst]: I would link to study template now

File: doc/overview/roadmap.rst [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+2. GUI elements to use PyQt5 (rather than TraitsUI/pyface)
+3. 3D plotting to use our abstracted 3D viz functions rather than Mayavi
+
+Once this is done, we can effectively switch to a PyVista backend.
+
+
+BIDS Integration
+^^^^^^^^^^^^^^^^
+
+MNE-Python should facilitate analyzing BIDS-compliant datasets. For more
```
Comment (agramfort) [inline doc/overview/roadmap.rst]:
```suggestion
MNE-Python should facilitate analyzing BIDS-compliant datasets thanks to
integration with the MNE-BIDS package. For more
```

File: doc/overview/roadmap.rst [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    1,200 healthy adult subjects. The available data includes MR structural
+    scans, behavioral data and (on a subset of the data) resting state and/or
+    task MEG data.
+- `Kymata Datasets <https://kymata-atlas.org/datasets>`__.
+    Current and archived EMEG measurement data, used to test hypotheses in the
+    Kymata atlas. The participants are healthy human adults listening to the
+    radio and/or watching films, and the data is comprised of (averaged) EEG
+    and MEG sensor data and source current reconstructions.
+- `BrainSignals <http://www.brainsignals.de/>`__
+    A website that lists a number of MEG datasets available for download.
```
Comment (agramfort) [inline doc/overview/roadmap.rst]: I would not focus now on openneuro datasets.

**Pull request merged**